version: '3.8'

services:
  dynamodb-local:
    image: "amazon/dynamodb-local:latest"
    container_name: dynamodb-local
    hostname: dynamodb
    ports:
      - "8000:8000"
    volumes:
      - "./docker/dynamodb:/home/dynamodblocal/data"
    working_dir: /home/dynamodblocal
    command: "-jar DynamoDBLocal.jar -sharedDb -dbPath ./data"

  minio:
    image: quay.io/minio/minio:RELEASE.2021-10-10T16-53-30Z
    hostname: minio
    volumes:
      - ./docker/minio/data:/data
      - ./docker/minio/export:/export
      - ./docker/minio/policies:/policies
    ports:
      - "9000:9000"
      - "9001:9001"
    environment:
      MINIO_ROOT_USER: root
      MINIO_ROOT_PASSWORD: himitsu123
    command: server --console-address ":9001" /data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  stepfn:
    image: amazon/aws-stepfunctions-local:latest
    hostname: stepfn
    depends_on:
      - dynamodb
      - minio
    ports:
      - "8083:8083"
    environment:
      LAMBDA_ENDPOINT: http://host.docker.internal:3001
      DYNAMODB_ENDPOINT: http://dynamodb:8000

networks:
  ssb-handson:
    external: true
    name: ssb-handson
